From 7f2bd177ea7f7a65f75cc3e5d2625324f7881451 Mon Sep 17 00:00:00 2001
From: Thomas Loimer <thomas.loimer@tuwien.ac.at>
Date: Tue, 26 Dec 2023 20:25:27 +0100
Subject: [PATCH 6/9] Make --enable-versioning work in out-of-tree builds

After ./configure --enable-versioning, the current git commit is written
to the version string (e.g., 3.2.9-5-g6990). This is useful for
developers and hacking.
---
 Makefile.am       | 10 ----------
 configure.ac      |  6 ++++--
 update-version_m4 |  4 +++-
 3 files changed, 7 insertions(+), 13 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 309ef6c..872cb37 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -26,13 +26,3 @@ endif
 EXTRA_DIST = CHANGES NOTES update-version_m4
 
 MAINTAINERCLEANFILES = Makefile.in aclocal.m4 config.h.in configure
-
-if ENABLE_VERSIONING
-# The generated Makefile lets configure and config.h.in depend on configure.ac,
-# and on all .m4 files. If configure.ac is changed, configure is re-created,
-# then config.h, probably only after version.m4 is updated, and configure
-# is again recreated and run. Only a nuisance, though.
-version.m4: configure.ac Makefile.am \
-	fig2dev/*.c fig2dev/*.h fig2dev/dev/*.c fig2dev/dev/*.h
-	. ./update-version_m4
-endif
diff --git a/configure.ac b/configure.ac
index 50adbe5..9ceb99f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -156,7 +156,8 @@ AC_ARG_ENABLE(versioning, [AS_HELP_STRING([--enable-versioning],
 	[enable changing the version number, only useful for \
 	 hacking (default: disable)])],
     [],[enableval=no])dnl
-AM_CONDITIONAL([ENABLE_VERSIONING], [test "x$enableval" = xyes])dnl
+AS_IF([test "x$enableval" = xyes],dnl
+    [update_version='( cd $srcdir && . ./update-version_m4 )'])dnl
 
 AC_ARG_ENABLE(transfig, [AS_HELP_STRING([--enable-transfig],
 	[build the transfig program (default: do not build)])],
@@ -336,7 +337,8 @@ AC_CONFIG_FILES([Makefile
 		fig2dev/tests/atlocal
 		fig2dev/tests/data/patterns.svg
 		fig2dev/tests/data/fillswclip.svg
-		transfig/Makefile])
+		transfig/Makefile],
+	[],[$update_version])
 
 AC_OUTPUT
 
diff --git a/update-version_m4 b/update-version_m4
index d77a8fa..c71528a 100755
--- a/update-version_m4
+++ b/update-version_m4
@@ -22,7 +22,9 @@
 # Author: Thomas Loimer <thomas.loimer@tuwien.ac.at>, 2016
 #
 
-if (git --version) >/dev/null 2>&1 && test "`git describe --abbrev=1`" != \
+# test, whether git is available, and we are in a git tree
+if git config --local -l >/dev/null 2>&1 &&	\
+    test "`git describe --abbrev=1`" !=		\
 	"`sed -n '/^m4_def.*FIG_VERS/ {s/.*, *\[\(.*\)\])/\1/;p}' version.m4`"
 then
     sed  -i "\
-- 
2.34.1

